mono/utils/mono-threads-coop.c

```diff
@@ -412,6 +420,11 @@ mono_threads_enter_gc_unsafe_region_unbalanced_with_info (MonoThreadInfo *info,
 		info->thread_saved_state [SELF_SUSPEND_STATE_INDEX].valid = FALSE;
 		break;
 	case AbortBlockingWait:
+	case AbortBlockingNotifyAndWait:
+		// in full coop, notify and wait doesn't need to notify
+		if (result == AbortBlockingNotifyAndWait && mono_threads_is_hybrid_suspension_enabled ()) {
```

Comment:
We should probably just duplicate the code for `AbortBlockingWait` and `AbortBlockingNotifyAndWait` to keep it consistent across the board.